docs/development.md

```diff
@@ -348,9 +349,82 @@ In the vagrant box, the dynflow repository is cloned to `/home/vagrant/dynflow`.
 The testing tool [smoker](https://github.com/theforeman/smoker) can be set up with the `centos7-foreman-smoker` box and tests can be run against a separate Foreman/Katello instance.
 
 To use:
-1. Ensure that you have a running instance of Foreman/Katello.  
+1. Ensure that you have a running instance of Foreman/Katello.
 2. Follow the example box definition in `vagrant/boxes.d/99-local.yaml.example` for `centos7-foreman-smoker` and update the `smoker_base_url` variable. With `pytest_run_tests` set to false, smoker tests will not be run by the playbook, but the box will be set up with pytest and the smoker repository will be cloned to the `vagrant` user's home directory.
 3. Run `vagrant up centos7-foreman-smoker`. A debug message will print showing the command to run smoker tests and the alias that has been set up. The alias is defined in `~/.bash_profile` on the box itself.
 4. You can then ssh into the smoker box. Ensure the hostname of the Foreman/Katello instance can be reached by the smoker box.
 5. From the smoker box, run tests as the vagrant user using the alias or running pytest commands manually. To change the testing options, please see [the smoker documentation](https://github.com/theforeman/smoker) and modify the alias or manually run pytest commands as necessary.
 
+## Advanced features
+
+These are playbooks, roles and variables provided to use additional funcionalities to test advanced foreman features.
+This list should be organized by foreman feature.
+
+* [Kerberos](#foreman-with-active-directory)
+
+### Foreman with Active Directory
+
+#### Prepare AD controller
+
+You need the AD controller with domain controller you want to test this against.
+Easiest is to install Windows server (ideally core to save resources, but its harder to install AD and setup DNS).
+This controller doesn't need to be searchable in your environmnet, we will point libvirt there.
+
+I will be using the domain ```example.com``` and expect your AD controller runs on IP ```192.168.1.8```.
```

Comment:
```suggestion
I will be using the domain ```example.com``` and expect your AD controller runs on IP ```192.0.2.8```.
```

[rfc5735](https://tools.ietf.org/html/rfc5735) ;)

Reply:
And I think you want singe backticks here, not three?

Reply:
Yeah, I guess I can, no idea why I used three, it was bit late and probably I went with three is more then one :thinking: :shrug: